File: src/main/java/net/lifecity/mc/skillmaster/user/SkillUserList.java
package net.lifecity.mc.skillmaster.user;

import org.bukkit.entity.Player;

import java.util.ArrayList;

/**
 * スキルユーザーを管理するためのリストクラス
 */
public class SkillUserList extends ArrayList<SkillUser> {

    /**
     * PlayerのUUIDが一致するSkillUserを取得します
     * @param player このプレイヤーのUUIDを使用します
     * @return UUIDが一致したSkillUserまたはnull
     */
    public SkillUser get(Player player) {
        for (SkillUser user : this) {
            if (user.match(player))
                return user;
        }
        return null;
    }

    /**
     * Playerが一致するSkillUserがこのリストに入っているか
     * @param player 対象となるPlayer
     * @return リストにPlayerが入っているか
     */
    public boolean contains(Player player) {
        return get(player) != null;
    }

    /**
     * プレイヤーをSkillUserとしてリストに追加します
     * @param player 追加するPlayer
     */
    public void add(Player player) {
        if (!contains(player)) {
            add(new SkillUser(player));
        }
    }

    /**
     * Playerが一致するSkillUserを削除します
     * @param player 対象となるPlayer
     */
    public void remove(Player player) {
        SkillUser user = get(player);
        if (user != null) remove(user);
    }
}

File: src/main/java/net/lifecity/mc/skillmaster/user/SkillUserList.kt
package net.lifecity.mc.skillmaster.user

import net.lifecity.mc.skillmaster.SkillUserNotFoundException
import org.bukkit.entity.Player

class SkillUserList {
    val list = mutableListOf<SkillUser>()

    /**
     * PlayerからSkillUserを特定します
     * @param player プレイヤー
     * @return 特定されたSkillUser
     */
    operator fun get(player: Player): SkillUser {
        for (user in list) {
            if (user.player == player)
                return user
        }
        throw SkillUserNotFoundException("SkillUser:${player.name} is not found")
    }

    /**
     * playerが格納されているかを返します
     * @param player プレイヤー
     * @return 格納されていたらtrue
     */
    fun contains(player: Player): Boolean {
        for (user in list) {
            if(user.player == player) return true
        }
        return false
    }

    /**
     * プレイヤーをSkillUserとして登録します
     * @param player プレイヤー
     */
    fun add(player: Player) {
        if (!contains(player))
            list.add(SkillUser(player))
    }

    /**
     * playerからSkillUserを特定し、格納されていたらリストから削除します
     * @param player プレイヤー
     */
    fun remove(player: Player) {
        val user = get(player)
        list.remove(user)
    }
}